assets/js/app.js

```diff
@@ -0,0 +1,55 @@
+let productsArray = [];
+
+function collectData() {
+    const productName = document.getElementById('productName').value
+    const productPrice = document.getElementById('productPrice').value
+    const productCategory = document.getElementById('productCategory').value
+    const productPhotoURL = document.getElementById('productPhotoURL').value
+    return {
+        productName: productName,
+        productPrice: productPrice,
+        productCategory: productCategory,
+        productPhotoURL: productPhotoURL,
+    }
+}
+
+function generateHTML(data, index) { // saves on newHTML the data by the mission requirements 
+    const newHTML =
+        `<tr>
+            <td>${data.productName}</td>
+            <td>${data.productPrice}</td>
+            <td>${data.productCategory}</td>
+            <td><img src="${data.productPhotoURL}"></td>
+            <td><button onclick="deleteProduct(${index})" class="btn btn-danger">Delete</button></td>
+        </tr>`
+    return newHTML
+}
+
+function renderAllProducts() { //render all product from the array
+    const productContainer = document.getElementById('productContainer')
+    productContainer.innerHTML = '' // Clear current display
+    for (let i = 0; i < productsArray.length; i++) {
+        const newHTML = generateHTML(productsArray[i], i)
+        productContainer.innerHTML += newHTML // adds each product from the array to the table
```

Comment:
modifying the DOM in a loop is a bad practice. You force to browser to keep re-rendering on each assignment. you should create a string variable and use `+=` on that variable and only after the loop ends you should modify the DOM *once*